persistence/remote/src/main/java/org/infinispan/persistence/remote/RemoteStore.java

```diff
@@ -303,13 +303,12 @@ private Object getValue(MarshallableEntry entry) {
 
    @Override
    public CompletionStage<Void> bulkUpdate(Publisher<MarshallableEntry<? extends K, ? extends V>> publisher) {
-      CompletableFuture<Void> future = new CompletableFuture<>();
-      Flowable.fromPublisher(publisher)
+      return Flowable.fromPublisher(publisher)
             .buffer(configuration.maxBatchSize())
+            // TODO: this is blocking
```

Comment:
`putAllAsync`?

Reply:
It was more to just point it out for future. This needs to be totally rewritten to only send a batch at at time. We can't unfortunately just replace with `putAllAsync` as we will end up firing multiple putAlls in parallel, We should really submit one and then fire the next after that one is complete.

Reply:
This needs to be reevaluated after we have a full and proper async SPI though. As the store as of this PR will be invoked in a persistence thread. We would rather not have to use a persistence thread and context switch as this store can do non blocking operations.